openmdao/core/indepvarcomp.py

```diff
@@ -48,6 +50,49 @@ def __init__(self, name=None, val=1.0, **kwargs):
                 raise ValueError("IndepVarComp init: '%s' is not supported "
                                  "in IndepVarComp." % illegal)
 
+    def initialize(self):
+        """
+        Declare options.
+        """
+        opt = self.options
+        opt.declare('name', types=str,
+                    desc="Name of the variable in this component's namespace.")
+        opt.declare('val', types=(float, list, tuple, np.ndarray), default=1.0,
+                    desc="The initial value of the variable being added in user-defined units.")
+        opt.declare('shape', types=(int, tuple, list), default=None,
+                    desc="Shape of this variable, only required if val is not an array.")
+        opt.declare('units', types=str, default=None,
+                    desc="Units in which the output variables will be provided to the "
+                         "component during execution.")
+        opt.declare('res_units', types=str, default=None,
+                    desc="Units in which the residuals of this output will be given to "
+                         "the user when requested.")
+        opt.declare('desc', types=str,
+                    desc="Description of the variable")
+        opt.declare('lower', types=(int, float, list, tuple, np.ndarray), default=None,
+                    desc="Lower bound(s) in user-defined units. It can be (1) a float, "
+                         "(2) an array_like consistent with the shape arg (if given), or "
```

Comment:
Many of these options are going to go away based on channel discussion (in particular, solver-related options like the bounds and scaling.)  Guess that just means you did a little extra work....

Reply:
yeah.. this was easy enough without resolving those issues right now